agent/src/beerocks/slave/agent_ucc_listener.cpp

```diff
@@ -108,6 +108,27 @@ bool agent_ucc_listener::handle_dev_get_param(std::unordered_map<std::string, st
             return false;
         value = std::string("aLid,") + alid;
         return true;
+    } else if (params["parameter"] == "macaddr") {
+        if (params.find("ruid") == params.end()) {
+            value += " missing ruid";
+            return false;
+        }
+        if (params.find("ssid") == params.end()) {
+            value += " missing ssid";
+            return false;
+        }
+        auto ruid = network_utils::mac_to_string(std::stoull(params["ruid"], nullptr, 16));
+        auto ssid = params["ssid"];
+
+        value += " macaddr not found for ruid " + ruid + " ssid " + ssid;
+        if (vaps_map.find(ruid) == vaps_map.end())
+            return false;
+        for (auto vap : vaps_map[ruid].vaps) {
+            if (std::string(vap.ssid) == ssid) {
+                value = "macaddr," + network_utils::mac_to_string(vap.mac);
+                return true;
+            }
+        }
```

Comment:
I think it will be more readable and clear like that:
```cpp
// `value` is empty
auto ruid_it = vaps_map.find(ruid);
if (vaps_map.find(ruid) != vaps_map.end())
    for (const auto& vap : ruid_it->vaps) {
        if (std::string(vap.ssid) == ssid) {
            value = "macaddr," + network_utils::mac_to_string(vap.mac);
            return true;
        }
    }
}
value += " macaddr not found for ruid " + ruid + " ssid " + ssid;
return false;
```
(nitpick)